### DISCUSSION TITLE
Ssh Aktivierung setzt Passwort nicht ...

### DISCUSSION THREAD

**Discussion by mblascht** [OpenCCU/OpenCCU#1922] [🇩🇪 Fragen&Antworten] [answered]
Hallo zusammen,

ich habe auf einer neuen Sd Karte das System mit der neuesten Version 3.65.8.20220831 aufsetzten müssen (alte Sd Karte war wohl defekt und hat nicht mehr gebootet...).

Ssh habe ich aktiviert und mein Pw gesetzt, nach dem erfolgreichen Ssh Login habe ich auf der Homematic-Raspi Shell dann den gemounteten Usb Stick mit den Backups finden können, Backup habe ich eingespielt und dann gebootet.
Nach dem erfolgreichen Restore (uff, mein Konfiguration war wieder da) geht aber den Ssh Zugriff nicht mehr.
Fehlermeldung ist ein wie ein falsches Passwort:
```
(base) mblascht@MacBook-MB ~ % ssh root@homematic-raspi        
root@homematic-raspi's password: 
Permission denied, please try again.
```
Versuchsweise habe in der ResperryMatic Oberfläche Ssh deaktiviert. Dann kommt die erwartet Fehlermeldung:
```
(base) mblascht@MacBook-MB ~ % ssh root@homematic-raspi
ssh: connect to host homematic-raspi port 22: Operation timed out
```
Nach Aktivierung von Ssh und setzen meines Pw kommt aber wieder die obige Fehlermeldung. So als ob Ssh Aktivierung funktioniert aber das Pw setzen nicht...

Auf dem alten System (auf der defekten Sd Karte) hatte ich die gleiche Symptomatik, habe dort den Ssh Login vor Monaten auch nicht (mehr) hinbekommen. 

Sieht so aus, als hätte ich die alte Symptomatik wieder restored. Hat jemand eine Idee, wo ich suchen könnte um wieder einen Ssh Zugriff hinzubekommen?

Beste Grüße

**Comment by jens-maus** [comment]
Hat dein gewähltes Passwort irgendwelche Sonderzeichen oder Umlaute? Wenn ja, mal ohne probieren.

**Comment by mblascht** [comment]
Nein, nur Kleinbuchstaben... Hab's eben noch mal neu gesetzt und einen Login versucht.
In der heruntergeladenen Logdatei finde ich nur den fehlgeschlagegen Login Versuch:
```
Sep  4 07:48:15 homematic-raspi auth.info sshd[20862]: Failed password for root from 172.16.16.24 port 53821 ssh2
```
Soll ich mal den Level erhöhen und dort weiter suchen?

**Comment by mblascht** [comment, marked as answer]
Habe den Loglevel erhöht, ssh deaktiviert, gebootet.
Nach Neustart, ssh+Passwort aktiviert und Loginversuch:
```
(base) mblascht@MacBook-MB Downloads % egrep 'user|auth' homematic-raspi-2022-09-04.log | grep -v multimac
...
Sep  4 08:44:21 homematic-raspi user.info firewall: configuration set
Sep  4 08:44:22 homematic-raspi auth.info sshd[3628]: Server listening on 0.0.0.0 port 22.
Sep  4 08:44:22 homematic-raspi auth.info sshd[3628]: Server listening on :: port 22.
Sep  4 08:44:38 homematic-raspi auth.err sshd[3673]: error: Could not get shadow information for root
Sep  4 08:44:38 homematic-raspi auth.info sshd[3673]: Failed password for root from 172.16.16.24 port 54126 ssh2
Sep  4 08:44:42 homematic-raspi auth.info sshd[3673]: Connection closed by authenticating user root 172.16.16.24 port 54126 [preauth]

**Reply by jens-maus** [reply]
Was du noch probieren kannst ist, Monitor+Tastatur anschließen, dann dort versuchen einzuloggen. Wenn auch das nicht geht vermute ich es liegt an der `/usr/local/etc/config/shadow` Datei bzw. es kann die vllt. gar nicht schreiben/ändern und deshalb auch diese `Could not get shadow information for root` Meldungen.

**Reply by mblascht** [reply]
nee, geht per Tastatur genauso nicht ... Per Script ermittelt über UI:
system.Exec ("ls -al /usr/local/etc/config/shadow", &stdout, &stderr) liefert
```
-rw-------    1 root     root            29 Sep  4 08:44 /usr/local/etc/config/shadow
```
und system.Exec ("cat /usr/local/etc/config/shadow", &stdout, &stderr);
```
avahi:*:::::::
sshd:*:::::::
```
Das root Passwort fehlt hier, jepp.
Was tun??

**Reply by jens-maus** [reply]
Nun, das sieht komplett "fishy" aus. Sind das wirklich die einzig beiden Einträge der shadow datei? Eigentlich sollte diese wie folgt bzw. so ähnlich aussehen:

```
root:$6XXXXXXXXXXX:17640:0:99999:7:::
daemon:*:10933:0:99999:7:::
bin:*:10933:0:99999:7:::
sys:*:10933:0:99999:7:::
sync:*:10933:0:99999:7:::
mail:*:10933:0:99999:7:::
www-data:*:10933:0:99999:7:::
halt:*:10933:0:99999:7:::
uucp:*:10933:0:99999:7:::
operator:*:10933:0:99999:7:::
ftp:*:10933:0:99999:7:::
nobody:*:10933:0:99999:7:::
_ntp:*:::::::
sshd:*:::::::
```

Hier scheint die Datei wohl kaputt gegangen zu sein und deshalb kann die WebUI das Passwort auch nicht setzen. Probier nun mal die `/etc/config_templates/shadow` Datei mit dem folgenden Kommando an die `/usr/local/etc/config/` stelle zu kopieren:

```
cp -a /etc/config_templates/shadow /usr/local/etc/config/
```

Danach dann nochmal via WebUI probieren das Passwort zu setzen und dann mal sehen was passiert.

**Reply by mblascht** [reply]
Danke. Bin wieder drin per ssh. 

Bin mir unsicher ob das Restore nicht noch mehr kaputt gemacht hat.
Sind die *sbk Dateien lesbar und veränderbar?
Ich habe nur ca. 10 x Thermostate/Fensterkontakte in Räume und HomeKit konfiguriert. Keine Scripte...
Ist es sinnvoll diese Einstellungen aus den sbk Dateien zu extrahieren, und nur diese auf einem frisch installierten System zu restoren?

**Reply by jens-maus** [reply]
> Danke. Bin wieder drin per ssh.
> 
> Bin mir unsicher ob das Restore nicht noch mehr kaputt gemacht hat.

Nun, in der Tat "fishy" was genau bzw. woher diese kaputte `shadow` Datei kommt?!? Vor allem der `avahi` Eintrag macht mich sehr stutzig, denn avahi wird bei RaspberryMatic gar nicht benutzt. Bist du von piVCCU oder einem anderen CCU System einmal umgestiegen? Weil kann sein das du dir das irgendwie eingeschleppt hast.

> Sind die *sbk Dateien lesbar und veränderbar? Ich habe nur ca. 10 x Thermostate/Fensterkontakte in Räume und HomeKit konfiguriert. Keine Scripte... Ist es sinnvoll diese Einstellungen aus den sbk Dateien zu extrahieren, und nur diese auf einem frisch installierten System zu restoren?

Die *.sbk Datei sollte man nicht direkt einfach so editieren/modifizieren. Du solltest einfach mal überlegen woher denn diese shadow Datei kommen kann denn die sieht in der Tat so aus das du die irgendwie eingeschleppt hast und dir damit quasi ein Ei gelegt hattest. Von RaspberryMatic selbst kann die so nicht kommen und das Restore selbst kann die so auch nicht kaputt gemacht haben.

**Reply by mblascht** [reply]
Übrigens vielen Dank noch mal für deine Hilfe (... und das am Sonntag)
Ursprünglich habe ich das System auf einem Linux Server im Docker Container betrieben. Da ich noch einen Raspi übrig hatte, habe ich das dorthin migriert...per Restore..
Jedenfalls ist die ``shadow`` im Backup ebenfalls schon kaputt, hab die sbk Datei ausgepackt:
```
(base) mblascht@MacBook-MB config % pwd
/Users/mblascht/Downloads/usr/local/etc/config

(base) mblascht@MacBook-MB config % ls -al sh*
-rw-------  1 mblascht  staff  29 21 Jul 01:16 shadow
-rw-------  1 mblascht  staff  29 21 Jul 00:57 shadow-

(base) mblascht@MacBook-MB config % cat shadow
avahi:*:::::::
sshd:*:::::::

(base) mblascht@MacBook-MB config % cat shadow-
avahi:*:::::::
sshd:*:::::::
```

Selektives Restore aus der sbk Datei ist wirklich keine Option?

**Reply by jens-maus** [reply]
... dann wir das Problem daher kommen das du ein Backup aus dee Docker Umgebung genutzt hast.

**Comment by Baxxy13** [comment]
Hast du vielleicht eine `/usr/local/etc/ssh/sshd_config` mit "komischen Einstellungen" in deinem System?

**Reply by mblascht** [reply]
Die /usr/local/etc/ssh/sshd_config gibt's bei mir nicht:
```
string stdout;
string stderr;
system.Exec ("ls -al /usr/local/etc/ssh/sshd_config", &stdout, &stderr);
WriteLine(stdout);
WriteLine(stderr);
```
```
ls: /usr/local/etc/ssh/sshd_config: No such file or directory
```
